### PR TITLE
Import all auth providers

### PR DESCRIPTION
**Pull request by JoelSpeed**
This change makes sure to import all available auth providers from `client-go`. This will make local development easier as people will be able to build their controllers and talk to clusters using non GCP authentication (eg OIDC, Azure).

Discussed initially on slack with @droot

### CODE REVIEW

File: pkg/scaffold/project/gopkg.go [hunk 1]
```diff
@@ -327,6 +327,11 @@ version="kubernetes-1.10.0"
 name="k8s.io/client-go"
 version="kubernetes-1.10.1"
 
+# Required version for client-go 1.10
+[[override]]
+name = "github.com/Azure/go-autorest"
+version = "9.1.0"
+
```
Comment (droot) [inline pkg/scaffold/project/gopkg.go]: We will soon be migrating to k8s 1.11 (coming Tuesday). Will this override go away with that ?

**Comment by JoelSpeed** [comment]
@droot I've rebased this onto master and the 1.11 changes, the override has indeed gone away now

**Comment by droot** [comment]
Awesome. @mengqiy travis tests are failing. seems like gosec related. Do we need to disable them ?

**Comment by droot** [comment]
Given that `pkg/scaffold` has been moved to kubebuilder to streamline the development, this PR needs to be reworked in the kubebuilder repo, so I am closing it in this repo.

**Pull request closed without merging**